Formula/fish.rb

```diff
@@ -22,9 +22,16 @@ class Fish < Formula
 
   def install
     system "autoconf" if build.head? || build.devel?
+    args = %W[
+      --prefix=#{prefix}
+      --with-extra-functionsdir=#{HOMEBREW_PREFIX}/share/fish/vendor_functions.d
```

Comment:
I think we'd be willing to change (though it won't be in until the next release; you could carry a patch until then?) but I'm not sure whether fish should just never create the directory (probably breaks some other packaging systems) or use `install || true` to soft-fail if the prefix is unwriteable; probably the latter.